src/async_fuse/memfs/kv_engine/mod.rs

```diff
@@ -225,8 +230,20 @@ pub trait KVEngine: Send + Sync + Debug + Sized {
     /// Lease grant
     async fn lease_grant(&self, ttl: i64) -> DatenLordResult<i64>;
 
+    /// Campaign leader
+    async fn campaign(&self, key: &KeyType, val: &ValueType, lease_id: i64) -> bool;
+
+    /// Lease keep alive
+    async fn create_keep_alive_session(&self, lease_id: i64, ttl: i64) -> Arc<Self::Session>;
+
     /// Range get, return all key-value pairs start with prefix
     async fn range(&self, prefix: &KeyType) -> DatenLordResult<Vec<ValueType>>;
+
+    /// Watch the key, return a receiver to receive the value
+    async fn watch(
+        &self,
+        key: &KeyType,
+    ) -> DatenLordResult<Arc<mpsc::Receiver<(String, Option<ValueType>)>>>;
```

Comment:
The comments for the returned tuple need to be improved.

Reply:
Okay, is it necessary to implement a struct to contain these receiver?

Reply:
If watch() is not used frequently, I think it is unnecessary.

Reply:
I think mpsc::Recevier should not be wrapped in Arc.